extension/src/format/sexpression.ts

```diff
@@ -2,94 +2,353 @@ import { closeParenStyle, maximumLineChars, longListFormatStyle, indentSpaces }
     symbLine(last: boolean = true): number {
         if (last) {
             let internalLines = 0;
-            if (this.symbol.startsWith(";|")) {
+            if (this.symbol.startsWith(';|')) {
                 for (let i = 0; i < this.symbol.length; i++) {
-                    if (this.symbol.charAt(i) == "\n") // it can handle the \r\n and \n
+                    if (this.symbol.charAt(i) === '\n') { // it can handle the \r\n and \n 
                         internalLines++;
+                    }
                 }
             }
             return this.line + internalLines;
         }
-        else return this.line;
+        else {
+            return this.line; 
+        }
     }
 
+
+    // Returns the length of the LispAtom's text value
     length(): number {
         return this.symbol.length;
     }
-    format(startColumn: number): string {
-        return this.symbol;
-    }
 
+
+    // Tests if the LispAtom is representing a single-line comment
     isLineComment(): boolean {
-        if (this.symbol.startsWith(";")
-            && !this.symbol.startsWith(";|"))
-            return true;
-        return false;
+		return this.symbol.startsWith(';') && !this.symbol.startsWith(';|');
     }
+	
 
+    // Tests if the LispAtom is representing any type of comment
     isComment(): boolean {
-        if (this.symbol.startsWith(";"))
-            return true;
-        return false;
+		return this.symbol.startsWith(';');
     }
+	
 
+    // Tests if the LispAtom is representing a structural closing parenthesis
     isRightParen(): boolean {
-        if (this.symbol == ")")
-            return true;
-        return false;
+		return this.symbol === ')';
     }
+	
+
+    // Tests if the LispAtom is representing a structural opening parenthesis
     isLeftParen(): boolean {
-        if (this.symbol == "(")
-            return true;
-        return false;
+		return this.symbol === '(';
     }
-    isValidSexpr(): boolean {
-        return true;
+
+
+    // Returns true if this LispAtom encapsulates the provided Position
+    contains(position: Position): boolean {
+        return this.getRange().contains(position);
+    }
+
+
+    // Gets the full range of the LispAtom and is capable of handling multi line strings or comments
+    getRange(): Range {
+        let cLine = this.line;
+        let cColm = this.column;
+        const begin: Position = new Position(cLine, cColm);
+        for (let i = 0; i < this.symbol.length; i++) {
+            const ch = this.symbol[i];
+            if (ch === '\n') {
+                cLine += 1;
+                cColm = 0;
+            } else {
+                cColm += 1;
+            }
+        }
+        const close: Position = new Position(cLine, cColm);        
+        return new Range(begin.line, begin.character, close.line, close.character);
+    }
+
+}
+
+
+// A utility class to hold and work with a collection of LispAtom|LispContainers
+export class LispContainer extends LispAtom {
```

Comment:
In lisp language there are two concepts: atom and S-expression. I am thinking why we need this LispContainer class, and I see it is brother for Sexpression.
LispContainer means some atoms which are not in the same Sexpression? Or those atoms are in several random Sexpression?